sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/StorageAllowedHeadersAndQueries.java

```diff
@@ -0,0 +1,104 @@
+package com.azure.storage.common.implementation;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class StorageAllowedHeadersAndQueries {
+    public static class BlobHeadersAndQueries {
```

Comment:
Why are you introducing static inner classes? There doesn't appear to be any need for this.

Reply:
Try to make the section clear.

Reply:
Make the structure flat will be a little hard to read, as there might have more coming for queue-crypto, file-datalake. 
Make them into separate class is a little extra to me. 
Let me know if you have any other opinions.